lib/pacemaker/pcmk_sched_messages.c

```diff
@@ -31,11 +31,22 @@ log_resource_details(pe_working_set_t *data_set)
     int rc = pcmk_rc_ok;
     pcmk__output_t *out = NULL;
     const char* argv[] = { "", NULL };
+    GListPtr unames = NULL;
     pcmk__supported_format_t formats[] = {
         PCMK__SUPPORTED_FORMAT_LOG,
         { NULL, NULL, NULL }
     };
 
+    /* Build up a list of nodes that we are allowed to output information for.
+     * This is necessary because out->message for all the resource-related
+     * messages expects such a list, due to the `crm_mon --node=` feature.  Here,
+     * we just make it a list of all the nodes.
+     */
+    for (GListPtr iter = data_set->nodes; iter != NULL; iter = iter->next) {
+        pe_node_t *node = (pe_node_t *) iter->data;
+        unames = g_list_append(unames, strdup(node->details->uname));
+    }
```

Comment:
Let's use g_list_prepend() for efficiency. If they need to stay in order, do g_list_reverse() after the loop.

I am a little concerned about the performance hit of creating this list -- theoretically a cluster could have hundreds of nodes. Too bad there's not an easy way to distinguish "match any node" from "match no node". I do have one idea, not saying it's a good idea, but maybe pe__unames_with_tag() could return NULL if tag_name is NULL, and a list of one item "\_\_empty\_\_" if nothing matches.

Reply:
Is '*' a valid name for a node?  If not, we could use that to mean "match any node", an empty list could be "match no node", and anything else is just a list of nodes to match.

Reply:
No it isn't, that would work. In fact we already use that for stonith_admin -H '*'.